osu.Game/Screens/OnlinePlay/Match/RoomModSelectOverlay.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Game.Online.Rooms;
+using osu.Game.Overlays;
+using osu.Game.Overlays.Mods;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Mods;
+
+namespace osu.Game.Screens.OnlinePlay.Match
+{
+    public partial class RoomModSelectOverlay : UserModSelectOverlay
+    {
+        [Resolved]
+        private IBindable<PlaylistItem> selectedItem { get; set; } = null!;
+
+        [Resolved]
+        private RulesetStore rulesets { get; set; } = null!;
+
+        private readonly List<Mod> roomMods = new List<Mod>();
+
+        public RoomModSelectOverlay()
+            : base(OverlayColourScheme.Plum)
+        {
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            selectedItem.BindValueChanged(_ =>
+            {
+                roomMods.Clear();
+
+                if (selectedItem.Value is PlaylistItem item)
+                {
+                    var rulesetInstance = rulesets.GetRuleset(item.RulesetID)?.CreateInstance();
+                    Debug.Assert(rulesetInstance != null);
+                    roomMods.AddRange(item.RequiredMods.Select(m => m.ToMod(rulesetInstance)));
+                }
+
+                SelectedMods.TriggerChange();
+            });
+        }
+
+        protected override void UpdateOverlayInformation(IReadOnlyList<Mod> mods)
+            => base.UpdateOverlayInformation(roomMods.Concat(mods).ToList());
```

Comment:
This is sort-of bad as it results in an allocation for every single change in mod settings, especially for sliders with `TransferValueOnCommit = false`. Not sure how to easily fix though, could change it to an enumeration and make `BeatmapAttributesDisplay.Mods` a bindable list so it operates on the same list rather than accepting a new list every call.

Reply:
It's pretty painful but I don't see immediately see a good way on resolving that. A pretty bad way would be to have another bindable in `ModSelectOverlay`, something like `ActiveMods` or something (contrary to `SelectedMods`), which would normally be a copy of / bound to `SelectedMods`, but for this every change to either `SelectedMods` or `playlistItem` would cause `ActiveMods` to contain the sum of them, and all components showing "mod effects" of that would be bound to that rather than `SelectedMods` outright.

Dunno. Maybe worth exploring, maybe not. Not sure myself at this stage even. It's extra complexity, but we're already having to add complexity in this PR, so maybe that added amount is not too much...?

Reply:
I was against the idea of introducing another list of mods in a `ModSelectOverlay`, but it's slowly growing on me now that I have coded it up, alongside the name `ActiveMods` as well. 

I've pushed the refactor in this PR to make reviewing easier. If it turns out to be too much complication, I can easily revert back to where I've had this PR.